src/hearings/containers/request-hearing/hearing-link/hearing-link.component.ts

```diff
@@ -40,14 +44,30 @@ export class HearingLinkComponent extends RequestHearingPageFlow implements OnIn
       caseReference: this.caseId,
       hearingId: ''
     }));
-    this.hearingLinksSub = this.hearingStore.pipe(select(fromHearingStore.getHearingLinks)).subscribe(
-      hearingLinks => {
-        if (hearingLinks.serviceLinkedCases) {
-          this.linkedCases = hearingLinks.serviceLinkedCases;
-          this.showSpinner = false;
+    this.generateLinkedCasesWithReasonDescription();
+  }
+
+  public generateLinkedCasesWithReasonDescription(): void {
+    this.hearingsServiceSub = this.hearingsService.loadCaseLinkingReasonCodes().subscribe(reasons => {
+      this.hearingLinksSub = this.hearingStore.pipe(select(fromHearingStore.getHearingLinks)).subscribe(
+        hearingLinks => {
```

Comment:
this should be using the updated subscriber / observer pattern and there is a lot going on in here, consider refactoring out the inside of this into another function.

Reply:
This will be done as part of the PR https://github.com/hmcts/rpx-xui-webapp/pull/2630 which you already commented on and merged later into this.